test/mock_chroot.bats

```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/env bats
+
+set -euo pipefail
+
+load ../src/bats-mock
+
+teardown() {
+    rm -rf "${BATS_TMPDIR}"/bats-mock.$$.*
+}
+
+@test 'mock_chroot without argument creates directory with minimal set of commands' {
+  run mock_chroot
+  [[ "${status}" -eq 0 ]]
+  [[ -x "${output}/awk" ]]
+  [[ -x "${output}/basename" ]]
+  [[ -x "${output}/bash" ]]
+  [[ -x "${output}/cat" ]]
+  [[ -x "${output}/chmod" ]]
+  [[ -x "${output}/chown" ]]
+  [[ -x "${output}/cp" ]]
+  [[ -x "${output}/cut" ]]
+  [[ -x "${output}/date" ]]
+  [[ -x "${output}/env" ]]
+  [[ -x "${output}/dirname" ]]
+  [[ -x "${output}/getopt" ]]
+  [[ -x "${output}/grep" ]]
+  [[ -x "${output}/head" ]]
+  [[ -x "${output}/id" ]]
+  [[ -x "${output}/find" ]]
+  [[ -x "${output}/hostname" ]]
+  [[ -x "${output}/ln" ]]
+  [[ -x "${output}/ls" ]]
+  [[ -x "${output}/mkdir" ]]
+  [[ -x "${output}/mktemp" ]]
+  [[ -x "${output}/mv" ]]
+  [[ -x "${output}/pidof" ]]
+  [[ -x "${output}/readlink" ]]
+  [[ -x "${output}/rm" ]]
+  [[ -x "${output}/rmdir" ]]
+  [[ -x "${output}/sed" ]]
+  [[ -x "${output}/sh" ]]
+  [[ -x "${output}/sleep" ]]
+  [[ -x "${output}/sort" ]]
+  [[ -x "${output}/split" ]]
+  [[ -x "${output}/tail" ]]
+  [[ -x "${output}/tee" ]]
+  [[ -x "${output}/tempfile" ]]
+  [[ -x "${output}/touch" ]]
+  [[ -x "${output}/tty" ]]
+  [[ -x "${output}/uname" ]]
+  [[ -x "${output}/uniq" ]]
+  [[ -x "${output}/unlink" ]]
+  [[ -x "${output}/wc" ]]
+  [[ -x "${output}/which" ]]
+  [[ -x "${output}/xargs" ]]
```

Comment:
I guess this list is specific to your machine, so perhaps a reduced set of commands would be enough to provide a decent test, so that this will pass even if run inside a plain Alpine docker image.


suggestion: `date`, `ln`, `ls`, `sh`, `rm`

Reply:
Indeed, I just picked a set of commonly used commands and added those to the mocked chroot. The point of `mock_chroot` is to provide a default minimal system.

I don’t understand if you would like that I adapted the test, so it would pass on systems with a smaller footprint, or do you want a smaller minimal system?